### DISCUSSION TITLE
Suggestion to support multiple configs for different ID servers

### DISCUSSION THREAD

**Discussion by goodluck1982** [rustdesk/rustdesk#1873] [Q&A] [answered]
I use two rustdesk-servers, one at home and another at office.

When I am in office, I prefer to use to the server at  office, because the net speed is high in the big intranet of our institute. However, at present I have to refill the infromation about the ID server, relay server, and the key every time after the change of my positon. It's too inconvenient!

I suggest to support multiple configs for different ID servers. Possible solutions are:
(1) Support individual config of server information for each remote desktop.
(2) Provides multiple configs (or profiles) and a method of convenient swith among them.
(3) At least a command-line support to specify config file by user. Then one can write different startup scripts for different rustdesk-servers.

**Comment by dayvan** [comment]
Yes, this would be very useful.
In my case I have a home server through which I connect to my devices. However, I'm also the designated family tech support guy, so their devices do not connect through my home computer.
If I had two profiles/configs, one wold be for my home devices configured with my local server, and the other would be to connect to RDs of family and friends.
Thanks.

**Comment by jacotec** [comment]
Exactly the same here ... one private RustDesk server and one in the company. From my Home machine I need to connect to both worlds, which is currently not possible.

**Comment by rustdesk** [comment]
[Specify id/relay server as part of target id when connecting to another device (id@server) ](https://github.com/rustdesk/rustdesk/issues/2118)

**Comment by jacotec** [comment]
Personally, I like the approach in #2118. In my opinion the key could be fetched from DNS (TXT entry). OK, if someone gets my public key he could use my server for his RustDesk clients, but he could not connect to any machine without knowing the ID and password. And why should someone use "some other" alien server whithout being sure that this server will be still up in a few months and with the risk of loosing all his remote connections. Also the project is Open Source, there are public Rustdesk servers and getting a VM in a cloud to run the server costs $1-2 per month?

**Comment by goodluck1982** [comment]
The form `xxx@yyy` is also a good idea and the `yyy` part **does not have to be actual host address**,  as suggested in https://github.com/rustdesk/rustdesk/issues/2118#issuecomment-1315061795

In fact, if the `yyy` part can also a **config name**, it can solve the problem that can not store key.

That is, one can define different configs, such as
```
conf1:
server=....
realy=...
key=...
```
```
conf2:
server=....
realy=...
key=...
```

Then use `12345678@conf1` to use the server, relay, key, etc defined in `conf1`.

The client can first try to treat the `yyy` part as a config name. If it can not find a config with the name `yyy` then it treat `yyy` as a domain name or IP.

**Comment by goodluck1982** [comment]
Previously, with version 1.1.9, I used the customized file name   `rustdesk-host=xxx,key=xxxx.exe` to specify server and key. Although this can be done only under windows, it works anyway.  I installed them to `C:\Program Files\RustDesk`.  

However, after I installed **1.2.2** today, it **REMOVED**  my previous files such as  `rustdesk-host=xxx,key=xxxx.exe` in the directory  `C:\Program Files\RustDesk`. This forces me to configure them again and brings trouble to users.

So I don't think it is a good way to put the host and key in the file name for specifying server.

**Comment by rustdesk** [comment, marked as answer]
https://github.com/rustdesk/rustdesk/issues/6198#issuecomment-1794694273

**Comment by Joaquim3** [comment]
I have the same profil.

- 1 cloud server to help my different communities
- 1 local server to serve my own home network.

Every time, I need to switch to one or the other, I need to re-type the IP+the KEY.

**If there was a "Dropdown box", where I would select "IP 1" or "IP 2", it would a gain of time.**

Thanks a lot anyway for your dedicated work ! 😀
